### ISSUE TITLE
Bazel@HEAD cannot build Bazel on OS X

### ISSUE THREAD

**Issue by dhalperi** [bazelbuild/bazel#7575]
### Description of the problem / feature request:

At d03dcfb450a8f3717d81f9ecde5ac24ec53e042a, build bazel (with Bazel 0.23.0) and then use the newly-built `bazel` binary to build bazel again.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
brew upgrade bazelbuild/tap/bazel  # install bazel 0.23.0
B23=/usr/local/bin/bazel
git checkout d03dcfb450a8f3717d81f9ecde5ac24ec53e042a
${B23} build //src:bazel
BHEAD=bazel-bin/src/bazel
${BHEAD} build //src:bazel
```

outputs:

```
➜  bazel git:(master) ${BHEAD} //src:bazel                                                   
Starting local Bazel server and connecting to it...
INFO: Invocation ID: 5e0b72f0-aebb-4d1a-90dd-7408fd73d4c1
DEBUG: /private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/osx/xcode_configure.bzl:153:9: Invoking xcode-locator failed, return code 1, stderr: error: Error Domain=NSOSStatusErrorDomain Code=-10814 "kLSApplicationNotFoundErr: E.g. no application claims the file" , stdout: 
ERROR: no such package '@local_config_cc//': Traceback (most recent call last):
	File "/private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/cpp/cc_configure.bzl", line 55
		configure_osx_toolchain(repository_ctx, overriden_tools)
	File "/private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/cpp/osx_cc_configure.bzl", line 65, in configure_osx_toolchain
		run_xcode_locator(repository_ctx, paths["@bazel_tool..."])
	File "/private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/osx/xcode_configure.bzl", line 154, in run_xcode_locator
		fail(error_msg)
Invoking xcode-locator failed, return code 1, stderr: error: Error Domain=NSOSStatusErrorDomain Code=-10814 "kLSApplicationNotFoundErr: E.g. no application claims the file" , stdout: 
ERROR: Analysis of target '//src:bazel' failed; build aborted: no such package '@local_config_cc//': Traceback (most recent call last):
	File "/private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/cpp/cc_configure.bzl", line 55
		configure_osx_toolchain(repository_ctx, overriden_tools)
	File "/private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/cpp/osx_cc_configure.bzl", line 65, in configure_osx_toolchain
		run_xcode_locator(repository_ctx, paths["@bazel_tool..."])
	File "/private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/osx/xcode_configure.bzl", line 154, in run_xcode_locator
		fail(error_msg)
Invoking xcode-locator failed, return code 1, stderr: error: Error Domain=NSOSStatusErrorDomain Code=-10814 "kLSApplicationNotFoundErr: E.g. no application claims the file" , stdout: 
INFO: Elapsed time: 5.943s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (89 packages loaded, 606 targets configured)
➜
```

### What operating system are you running Bazel on?

OS X latest

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

➜  bazel git:(master) git remote get-url origin ; git rev-parse master ; git rev-parse HEAD
git@github.com:bazelbuild/bazel.git
d03dcfb450a8f3717d81f9ecde5ac24ec53e042a
d03dcfb450a8f3717d81f9ecde5ac24ec53e042a

###  Have you found anything relevant by searching the web?

No.

### Any other information, logs, or outputs that you want to share?

See #7397 for more info on my setup. One key thing: I have xcode-cli tools, but not XCode application.

The root cause is the named commit; problem does not happen at previous commit of ac6d4fd194c2d9aba13ce613ec373f627210b18d

**Comment by dhalperi** [comment]
cc: @keith

**Comment by keith** [comment]
Interesting. We thought this case should still work. Can you post the output of:

```
env -i xcrun xcodebuild -version ; echo $?
```

**Comment by keith** [comment]
For more context this is why this block was left unchanged to not `fail()` https://github.com/bazelbuild/bazel/blob/d03dcfb450a8f3717d81f9ecde5ac24ec53e042a/tools/osx/xcode_configure.bzl#L175-L187

**Comment by dhalperi** [comment]
➜  bazel git:(master) ✗ env -i xcrun xcodebuild -version ; echo $?

xcrun: error: unable to find utility "xcodebuild", not a developer tool or in PATH
72
➜  bazel git:(master) ✗ xcodebuild  
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

**Comment by dhalperi** [comment]
What does the normal value look like? I do have all these:

```
➜  bazel git:(master) ✗ xcrun --version
xcrun version 43.
➜  bazel git:(master) ✗ xcode-select --version
xcode-select version 2354.
➜  bazel git:(master) ✗ xcrun --show-sdk-version        
10.14.1
➜  bazel git:(master) ✗ xcrun --show-sdk-build-version
18B71
➜  bazel git:(master) ✗ xcrun --show-sdk-platform-version
xcrun: error: unable to lookup item 'PlatformVersion' from command line tools installation
xcrun: error: unable to lookup item 'PlatformVersion' in SDK '/'
```

**Comment by keith** [comment]
I think I understand the issue, I'll submit a fix shortly

**Comment by keith** [comment]
Here's the PR https://github.com/bazelbuild/bazel/pull/7576, I'm attempting to test the fix locally by forcing xcode_locator to fail

**Comment by dhalperi** [comment]
Works for me:

```
➜  bazel git:(master) git fetch origin pull/7576/head:test-fix
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 7 (delta 5), reused 7 (delta 5), pack-reused 0
Unpacking objects: 100% (7/7), done.
From github.com:bazelbuild/bazel
 * [new ref]               refs/pull/7576/head -> test-fix
➜  bazel git:(master) git checkout test-fix  
Switched to branch 'test-fix'
➜  bazel git:(test-fix) git merge master
Already up to date.
➜  bazel git:(test-fix) bazel build //src:bazel && cp -f bazel-bin/src/bazel ~/bin/ && bazel build //src:bazel
INFO: Invocation ID: 1b2bc812-8647-4b66-bf12-1a8945ff6e00
INFO: Analysed target //src:bazel (1 packages loaded, 51 targets configured).
INFO: Found 1 target...
Target //src:bazel up-to-date:
  bazel-bin/src/bazel
INFO: Elapsed time: 16.120s, Critical Path: 15.59s
INFO: 5 processes: 5 darwin-sandbox.
INFO: Build completed successfully, 7 total actions
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
INFO: Invocation ID: ef97620c-94fa-447d-836c-34b76eee8ea6
DEBUG: /private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/osx/xcode_configure.bzl:154:9: Invoking xcode-locator failed, return code 1, stderr: error: Error Domain=NSOSStatusErrorDomain Code=-10814 "kLSApplicationNotFoundErr: E.g. no application claims the file" , stdout: 
INFO: Analysed target //src:bazel (217 packages loaded, 10234 targets configured).
INFO: Found 1 target...
Target //src:bazel up-to-date:
  bazel-bin/src/bazel
INFO: Elapsed time: 23.933s, Critical Path: 2.45s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
```

**Comment by dhalperi** [comment]
(I tried again with `bazel clean` thrown in, got same successful result.)

**Comment by keith** [comment]
Have you always seen this warning:

```
DEBUG: /private/var/tmp/_bazel_dan/53ac6b78c42db31234d67d8ef02ecc08/external/bazel_tools/tools/osx/xcode_configure.bzl:154:9: Invoking xcode-locator failed, return code 1, stderr: error: Error Domain=NSOSStatusErrorDomain Code=-10814 "kLSApplicationNotFoundErr: E.g. no application claims the file" , stdout:
```

Even before my patch? I assume so, I just want to verify

**Comment by dhalperi** [comment]
Sorry, but I'm not sure. Usually there are so many warnings I phase them out...

e.g.,

```
INFO: From Executing genrule //tools/osx:xcode-locator-genrule:
ld: warning: text-based stub file /System/Library/Frameworks//CoreServices.framework/CoreServices.tbd and library file /System/Library/Frameworks//CoreServices.framework/CoreServices are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks//Foundation.framework/Foundation.tbd and library file /System/Library/Frameworks//Foundation.framework/Foundation are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks//Foundation.framework/Foundation.tbd and library file /System/Library/Frameworks//Foundation.framework/Foundation are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks//CFNetwork.framework/Versions/A/CFNetwork.tbd and library file /System/Library/Frameworks//CFNetwork.framework/Versions/A/CFNetwork are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/FSEvents.framework/Versions/A/FSEvents.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/FSEvents.framework/Versions/A/FSEvents are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/CarbonCore.framework/Versions/A/CarbonCore.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/CarbonCore.framework/Versions/A/CarbonCore are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/Metadata.framework/Versions/A/Metadata.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/Metadata.framework/Versions/A/Metadata are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/OSServices.framework/Versions/A/OSServices.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/OSServices.framework/Versions/A/OSServices are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SearchKit.framework/Versions/A/SearchKit.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SearchKit.framework/Versions/A/SearchKit are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks//CoreFoundation.framework/Versions/A/CoreFoundation.tbd and library file /System/Library/Frameworks//CoreFoundation.framework/Versions/A/CoreFoundation are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/AE.framework/Versions/A/AE.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/AE.framework/Versions/A/AE are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/LaunchServices.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/LaunchServices.framework/Versions/A/LaunchServices are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/DictionaryServices.framework/Versions/A/DictionaryServices.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/DictionaryServices.framework/Versions/A/DictionaryServices are out of sync. Falling back to library file for linking.
ld: warning: text-based stub file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SharedFileList.framework/Versions/A/SharedFileList.tbd and library file /System/Library/Frameworks/CoreServices.framework/Versions/A/Frameworks/SharedFileList.framework/Versions/A/SharedFileList are out of sync. Falling back to library file for linking.
Target //src:bazel up-to-date:
```

**Comment by keith** [comment]
Those are interesting, but definitely shouldn't be related to this change 🤔

**Comment by dhalperi** [comment]
Yes, did not mean to imply they were related. Just that I don't see warnings anymore ;)

**Comment by keith** [comment]
No worries! I pinged some reviewers on that PR so hopefully it is merged soon. Sorry about that!

**Comment by keith** [comment]
FYI https://github.com/bazelbuild/bazel/issues/7592

**Comment by aragos** [comment]
As @keith somewhat obliquely referred to above, this should be fixed by c6c9b58. A roll forward (tracked in #7592) will need to make sure to avoid this issue.

**Comment by keith** [comment]
Sorry I figured the context on the issue would explain the comment
